lib/tools/common/patching_parallel.py

```
@@ -0,0 +1,1395 @@
+
+        # Initialize commit_messages list for tracking progress
+        commit_messages = []
+
+        log.debug(f"[Worker {item.mount_path}] Processing patch {item.patch_index}: {patch_id}")
+
+        # STEP 1: Configure git safe.directory and reset git tree
+        # ONLY reset if this is the first patch in the sequence
+        # Subsequent patches build on the cumulative changes
+        try:
+            # Add mount path to git safe.directory to handle /tmp directories
+            subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)